### PR TITLE
 Add string equals filtering and indexing

### PR DESCRIPTION
**Pull request by Laremere**
Part of implementing #681

### CODE REVIEW

File: internal/statestore/redis_indices.go [hunk 1]
```diff
@@ -58,6 +58,8 @@ func extractIndexedFields(cfg config.View, t *pb.Ticket) map[string]float64 {
 				value = 1
 			}
 			result[boolIndexName(attribute)] = value
+		case *structpb.Value_StringValue:
```
Comment (sawagh) [inline internal/statestore/redis_indices.go]: Could you as a separate task document the usage of filters / expectations of what needs to be specified etc.? Some easy reference we can point folks to for the current indexing / filtering mechanism?

File: api/messages.proto [hunk 2]
```diff
@@ -97,6 +97,23 @@ message BoolEqualsFilter {
   bool value = 2;
 }
 
+// Filters strings exactly equaling a value.
+//   attribute: "foo"
```
Comment (sawagh) [inline api/messages.proto]: Can I specify value of my attribute to be null? Should we fail setting it?

Comment (Laremere) [reply api/messages.proto]: Added null to does not match list.

File: internal/statestore/redis_indices.go (hunk 1 above)
Comment (Laremere) [reply internal/statestore/redis_indices.go]: Forgot to add strings to the documentation above.  Are you expecting more?

Comment (sawagh) [reply internal/statestore/redis_indices.go]: I meant website documentation on usage - not so much the code documentation. Again, not for this PR - but something the user guide should have.

**Pull request merged**